### PR TITLE
Pin back pandas for ray-data

### PR DESCRIPTION
**Pull request by timkpaine**
xref: ray-project/ray#42842
xref: conda-forge/ray-packages-feedstock#147

Checklist

* [x] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [ ] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [ ] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [ ] Ran `python show_diff.py` and posted the output as part of the PR.
* [ ] Modifications won't affect packages built in the future. 

FYI @conda-forge/ray-packages 
Will conflict with https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/645 but I'll resolve

```
================================================================================
================================================================================
linux-armv7l
================================================================================
================================================================================
win-32
================================================================================
================================================================================
osx-arm64
osx-arm64::ray-data-2.9.0-py311ha1ab1f8_2.conda
osx-arm64::ray-data-2.9.1-py311ha1ab1f8_0.conda
osx-arm64::ray-data-2.9.0-py310hb6292c7_1.conda
osx-arm64::ray-data-2.9.0-py39hdf13c20_1.conda
osx-arm64::ray-data-2.9.1-py38h150bfb4_0.conda
osx-arm64::ray-data-2.9.0-py311ha1ab1f8_0.conda
osx-arm64::ray-data-2.9.1-py39hdf13c20_1.conda
osx-arm64::ray-data-2.9.0-py38h150bfb4_0.conda
osx-arm64::ray-data-2.9.1-py310hb6292c7_1.conda
osx-arm64::ray-data-2.9.1-py39hdf13c20_0.conda
osx-arm64::ray-data-2.9.0-py310hb6292c7_2.conda
osx-arm64::ray-data-2.9.0-py310hb6292c7_0.conda
osx-arm64::ray-data-2.9.0-py38h150bfb4_2.conda
osx-arm64::ray-data-2.9.0-py39hdf13c20_0.conda
osx-arm64::ray-data-2.9.0-py311ha1ab1f8_1.conda
osx-arm64::ray-data-2.9.1-py311ha1ab1f8_1.conda
osx-arm64::ray-data-2.9.1-py38h150bfb4_1.conda
osx-arm64::ray-data-2.9.1-py310hb6292c7_0.conda
osx-arm64::ray-data-2.9.0-py38h150bfb4_1.conda
osx-arm64::ray-data-2.9.0-py39hdf13c20_2.conda
-    "pandas >=1.3",
+    "pandas >=1.3,<2.2*",
================================================================================
================================================================================
linux-ppc64le
================================================================================
================================================================================
linux-aarch64
================================================================================
================================================================================
noarch
================================================================================
================================================================================
win-64
win-64::ray-data-1.8.0-py38haa244fe_0.tar.bz2
win-64::ray-data-1.9.2-py38haa244fe_0.tar.bz2
win-64::ray-data-1.12.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.11.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.11.1-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.13.0-py310h5588dad_2.tar.bz2
win-64::ray-data-1.13.0-py310h5588dad_1.tar.bz2
win-64::ray-data-1.13.0-py37h03978a9_2.tar.bz2
win-64::ray-data-1.10.0-py37h03978a9_0.tar.bz2
win-64::ray-data-1.9.2-py37h03978a9_0.tar.bz2
win-64::ray-data-1.12.0-py37h03978a9_0.tar.bz2
win-64::ray-data-1.8.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.13.0-py37h03978a9_0.tar.bz2
win-64::ray-data-1.13.0-py37h03978a9_1.tar.bz2
win-64::ray-data-1.12.1-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.11.0-py37h03978a9_0.tar.bz2
win-64::ray-data-1.10.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.12.0-py38haa244fe_0.tar.bz2
win-64::ray-data-1.11.0-py38haa244fe_0.tar.bz2
win-64::ray-data-1.8.0-py37h03978a9_0.tar.bz2
win-64::ray-data-1.13.0-py38haa244fe_0.tar.bz2
win-64::ray-data-1.13.0-py39hcbf5309_2.tar.bz2
win-64::ray-data-1.11.1-py38haa244fe_0.tar.bz2
win-64::ray-data-1.12.1-py37h03978a9_0.tar.bz2
win-64::ray-data-1.10.0-py38haa244fe_0.tar.bz2
win-64::ray-data-1.13.0-py38haa244fe_2.tar.bz2
win-64::ray-data-1.13.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.11.1-py37h03978a9_0.tar.bz2
win-64::ray-data-1.13.0-py39hcbf5309_1.tar.bz2
win-64::ray-data-1.13.0-py38haa244fe_1.tar.bz2
win-64::ray-data-1.9.2-py39hcbf5309_0.tar.bz2
win-64::ray-data-1.12.1-py38haa244fe_0.tar.bz2
-    "pandas",
+    "pandas >=1.3,<2.2*",
win-64::ray-data-2.6.3-py310h5588dad_0.conda
win-64::ray-data-2.2.0-py310h5588dad_2.conda
win-64::ray-data-2.6.3-py38haa244fe_1.conda
win-64::ray-data-2.6.3-py39hcbf5309_0.conda
win-64::ray-data-2.5.0-py38haa244fe_0.conda
win-64::ray-data-2.0.0-py38haa244fe_1.tar.bz2
win-64::ray-data-2.6.3-py38haa244fe_0.conda
win-64::ray-data-2.2.0-py310h5588dad_1.conda
win-64::ray-data-2.8.1-py311h1ea47a8_0.conda
win-64::ray-data-2.2.0-py39hcbf5309_2.conda
win-64::ray-data-2.4.0-py38haa244fe_0.conda
win-64::ray-data-2.3.0-py310h5588dad_0.conda
win-64::ray-data-2.0.0-py37h03978a9_1.tar.bz2
win-64::ray-data-2.6.3-py39hcbf5309_1.conda
win-64::ray-data-2.8.0-py310h5588dad_0.conda
win-64::ray-data-2.0.1-py39hcbf5309_0.tar.bz2
win-64::ray-data-2.7.1-py39hcbf5309_0.conda
win-64::ray-data-2.8.0-py311h1ea47a8_0.conda
win-64::ray-data-2.1.0-py38haa244fe_1.conda
win-64::ray-data-2.0.1-py310h5588dad_1.tar.bz2
win-64::ray-data-2.8.1-py38haa244fe_1.conda
win-64::ray-data-2.7.1-py38haa244fe_0.conda
win-64::ray-data-2.5.0-py310h5588dad_0.conda
win-64::ray-data-2.0.1-py38haa244fe_0.tar.bz2
win-64::ray-data-2.1.0-py39hcbf5309_0.conda
win-64::ray-data-2.2.0-py310h5588dad_0.conda
win-64::ray-data-2.6.1-py310h5588dad_0.conda
win-64::ray-data-2.6.2-py39hcbf5309_0.conda
win-64::ray-data-2.1.0-py39hcbf5309_1.conda
win-64::ray-data-2.8.0-py38haa244fe_1.conda
win-64::ray-data-2.0.0-py39hcbf5309_0.tar.bz2
win-64::ray-data-2.8.1-py38haa244fe_0.conda
win-64::ray-data-2.8.0-py311h1ea47a8_2.conda
win-64::ray-data-2.1.0-py38haa244fe_0.conda
win-64::ray-data-2.8.0-py310h5588dad_1.conda
win-64::ray-data-2.7.1-py310h5588dad_1.conda
win-64::ray-data-2.6.1-py38haa244fe_0.conda
win-64::ray-data-2.6.1-py39hcbf5309_0.conda
win-64::ray-data-2.6.2-py310h5588dad_0.conda
win-64::ray-data-2.8.0-py39hcbf5309_0.conda
win-64::ray-data-2.7.0-py39hcbf5309_0.conda
win-64::ray-data-2.4.0-py310h5588dad_0.conda
win-64::ray-data-2.9.1-py38haa244fe_1.conda
win-64::ray-data-2.8.1-py39hcbf5309_1.conda
win-64::ray-data-2.2.0-py38haa244fe_1.conda
win-64::ray-data-2.8.1-py39hcbf5309_0.conda
win-64::ray-data-2.9.1-py39hcbf5309_1.conda
win-64::ray-data-2.5.1-py39hcbf5309_0.conda
win-64::ray-data-2.8.0-py39hcbf5309_2.conda
win-64::ray-data-2.2.0-py38haa244fe_0.conda
win-64::ray-data-2.5.1-py38haa244fe_0.conda
win-64::ray-data-2.0.1-py310h5588dad_0.tar.bz2
win-64::ray-data-2.5.1-py310h5588dad_0.conda
win-64::ray-data-2.0.0-py310h5588dad_0.tar.bz2
win-64::ray-data-2.6.2-py38haa244fe_0.conda
win-64::ray-data-2.7.0-py310h5588dad_0.conda
win-64::ray-data-2.8.1-py310h5588dad_1.conda
win-64::ray-data-2.6.3-py310h5588dad_1.conda
win-64::ray-data-2.0.0-py38haa244fe_0.tar.bz2
win-64::ray-data-2.2.0-py39hcbf5309_0.conda
win-64::ray-data-2.5.0-py39hcbf5309_0.conda
win-64::ray-data-2.3.0-py39hcbf5309_0.conda
win-64::ray-data-2.8.0-py38haa244fe_0.conda
win-64::ray-data-2.8.1-py310h5588dad_0.conda
win-64::ray-data-2.9.1-py311h1ea47a8_1.conda
win-64::ray-data-2.0.0-py37h03978a9_0.tar.bz2
win-64::ray-data-2.0.1-py38haa244fe_1.tar.bz2
win-64::ray-data-2.1.0-py310h5588dad_0.conda
win-64::ray-data-2.8.0-py38haa244fe_2.conda
win-64::ray-data-2.7.1-py38haa244fe_1.conda
win-64::ray-data-2.1.0-py310h5588dad_1.conda
win-64::ray-data-2.2.0-py38haa244fe_2.conda
win-64::ray-data-2.8.0-py310h5588dad_2.conda
win-64::ray-data-2.7.1-py311h1ea47a8_0.conda
win-64::ray-data-2.2.0-py39hcbf5309_1.conda
win-64::ray-data-2.7.1-py311h1ea47a8_1.conda
win-64::ray-data-2.8.0-py311h1ea47a8_1.conda
win-64::ray-data-2.0.1-py39hcbf5309_1.tar.bz2
win-64::ray-data-2.7.1-py39hcbf5309_1.conda
win-64::ray-data-2.7.0-py38haa244fe_0.conda
win-64::ray-data-2.9.1-py310h5588dad_1.conda
win-64::ray-data-2.8.0-py39hcbf5309_1.conda
win-64::ray-data-2.4.0-py39hcbf5309_0.conda
win-64::ray-data-2.0.0-py310h5588dad_1.tar.bz2
win-64::ray-data-2.0.0-py39hcbf5309_1.tar.bz2
win-64::ray-data-2.7.1-py310h5588dad_0.conda
win-64::ray-data-2.8.1-py311h1ea47a8_1.conda
win-64::ray-data-2.3.0-py38haa244fe_0.conda
-    "pandas >=1.3",
+    "pandas >=1.3,<2.2*",
================================================================================
================================================================================
osx-64
osx-64::ray-data-2.8.0-py39h6e9494a_0.conda
osx-64::ray-data-2.9.1-py38h50d1736_1.conda
osx-64::ray-data-2.8.1-py310h2ec42d9_1.conda
osx-64::ray-data-2.9.1-py39h6e9494a_0.conda
osx-64::ray-data-2.7.1-py311h6eed73b_1.conda
osx-64::ray-data-2.9.1-py310h2ec42d9_0.conda
osx-64::ray-data-2.9.0-py38h50d1736_2.conda
osx-64::ray-data-2.8.1-py311h6eed73b_1.conda
osx-64::ray-data-2.7.1-py39h6e9494a_1.conda
osx-64::ray-data-2.8.0-py38h50d1736_2.conda
osx-64::ray-data-2.8.0-py311h6eed73b_0.conda
osx-64::ray-data-2.8.1-py310h2ec42d9_0.conda
osx-64::ray-data-2.9.0-py39h6e9494a_1.conda
osx-64::ray-data-2.9.0-py310h2ec42d9_1.conda
osx-64::ray-data-2.8.0-py310h2ec42d9_1.conda
osx-64::ray-data-2.9.0-py311h6eed73b_2.conda
osx-64::ray-data-2.8.0-py310h2ec42d9_2.conda
osx-64::ray-data-2.8.1-py38h50d1736_0.conda
osx-64::ray-data-2.8.1-py39h6e9494a_0.conda
osx-64::ray-data-2.8.1-py38h50d1736_1.conda
osx-64::ray-data-2.8.0-py311h6eed73b_1.conda
osx-64::ray-data-2.9.0-py310h2ec42d9_0.conda
osx-64::ray-data-2.8.0-py38h50d1736_1.conda
osx-64::ray-data-2.9.1-py39h6e9494a_1.conda
osx-64::ray-data-2.9.0-py311h6eed73b_0.conda
osx-64::ray-data-2.8.1-py311h6eed73b_0.conda
osx-64::ray-data-2.9.1-py310h2ec42d9_1.conda
osx-64::ray-data-2.8.0-py38h50d1736_0.conda
osx-64::ray-data-2.9.1-py311h6eed73b_1.conda
osx-64::ray-data-2.9.0-py39h6e9494a_0.conda
osx-64::ray-data-2.9.0-py38h50d1736_0.conda
osx-64::ray-data-2.8.0-py39h6e9494a_1.conda
osx-64::ray-data-2.9.0-py311h6eed73b_1.conda
osx-64::ray-data-2.9.0-py38h50d1736_1.conda
osx-64::ray-data-2.8.1-py39h6e9494a_1.conda
osx-64::ray-data-2.9.0-py39h6e9494a_2.conda
osx-64::ray-data-2.8.0-py310h2ec42d9_0.conda
osx-64::ray-data-2.9.1-py38h50d1736_0.conda
osx-64::ray-data-2.9.0-py310h2ec42d9_2.conda
osx-64::ray-data-2.8.0-py311h6eed73b_2.conda
osx-64::ray-data-2.7.1-py310h2ec42d9_1.conda
osx-64::ray-data-2.7.1-py38h50d1736_1.conda
osx-64::ray-data-2.8.0-py39h6e9494a_2.conda
osx-64::ray-data-2.9.1-py311h6eed73b_0.conda
-    "pandas >=1.3",
+    "pandas >=1.3,<2.2*",
================================================================================
================================================================================
linux-64
linux-64::ray-data-1.8.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.9.2-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.11.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.13.0-py39hf3d152e_2.tar.bz2
linux-64::ray-data-1.13.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.12.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.13.0-py38h578d9bd_1.tar.bz2
linux-64::ray-data-1.12.1-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.12.1-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.11.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.11.0-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.12.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.9.2-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.13.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.12.1-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.13.0-py310hff52083_2.tar.bz2
linux-64::ray-data-1.11.1-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.11.1-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.13.0-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.8.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.9.2-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.10.0-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.10.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.13.0-py310hff52083_1.tar.bz2
linux-64::ray-data-1.13.0-py37h89c1867_1.tar.bz2
linux-64::ray-data-1.10.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-1.11.1-py37h89c1867_0.tar.bz2
linux-64::ray-data-1.8.0-py38h578d9bd_0.tar.bz2
linux-64::ray-data-1.13.0-py39hf3d152e_1.tar.bz2
linux-64::ray-data-1.13.0-py37h89c1867_2.tar.bz2
linux-64::ray-data-1.13.0-py38h578d9bd_2.tar.bz2
linux-64::ray-data-1.12.0-py38h578d9bd_0.tar.bz2
-    "pandas",
+    "pandas >=1.3,<2.2*",
linux-64::ray-data-2.3.0-py310hff52083_0.conda
linux-64::ray-data-2.8.0-py310hff52083_2.conda
linux-64::ray-data-2.8.0-py38h578d9bd_0.conda
linux-64::ray-data-2.7.1-py310hff52083_0.conda
linux-64::ray-data-2.0.1-py38h578d9bd_1.tar.bz2
linux-64::ray-data-2.0.1-py310hff52083_0.tar.bz2
linux-64::ray-data-2.0.0-py39hf3d152e_0.tar.bz2
linux-64::ray-data-2.6.3-py39hf3d152e_1.conda
linux-64::ray-data-2.6.1-py38h578d9bd_0.conda
linux-64::ray-data-2.9.1-py311h38be061_1.conda
linux-64::ray-data-2.9.0-py38h578d9bd_0.conda
linux-64::ray-data-2.0.1-py39hf3d152e_1.tar.bz2
linux-64::ray-data-2.5.0-py310hff52083_0.conda
linux-64::ray-data-2.8.1-py310hff52083_1.conda
linux-64::ray-data-2.9.0-py311h38be061_0.conda
linux-64::ray-data-2.7.1-py39hf3d152e_0.conda
linux-64::ray-data-2.4.0-py38h578d9bd_0.conda
linux-64::ray-data-2.0.0-py310hff52083_0.tar.bz2
linux-64::ray-data-2.6.2-py38h578d9bd_0.conda
linux-64::ray-data-2.7.0-py310hff52083_0.conda
linux-64::ray-data-2.4.0-py310hff52083_0.conda
linux-64::ray-data-2.3.0-py38h578d9bd_0.conda
linux-64::ray-data-2.0.0-py310hff52083_1.tar.bz2
linux-64::ray-data-2.6.3-py39hf3d152e_0.conda
linux-64::ray-data-2.1.0-py38h578d9bd_1.conda
linux-64::ray-data-2.7.1-py38h578d9bd_0.conda
linux-64::ray-data-2.8.0-py39hf3d152e_1.conda
linux-64::ray-data-2.9.0-py311h38be061_1.conda
linux-64::ray-data-2.2.0-py38h578d9bd_0.conda
linux-64::ray-data-2.6.1-py39hf3d152e_0.conda
linux-64::ray-data-2.8.0-py311h38be061_1.conda
linux-64::ray-data-2.2.0-py39hf3d152e_1.conda
linux-64::ray-data-2.0.0-py38h578d9bd_0.tar.bz2
linux-64::ray-data-2.9.1-py39hf3d152e_1.conda
linux-64::ray-data-2.0.1-py39hf3d152e_0.tar.bz2
linux-64::ray-data-2.1.0-py39hf3d152e_1.conda
linux-64::ray-data-2.9.1-py310hff52083_1.conda
linux-64::ray-data-2.9.0-py38h578d9bd_1.conda
linux-64::ray-data-2.0.0-py39hf3d152e_1.tar.bz2
linux-64::ray-data-2.9.0-py310hff52083_1.conda
linux-64::ray-data-2.0.0-py37h89c1867_1.tar.bz2
linux-64::ray-data-2.9.1-py38h578d9bd_1.conda
linux-64::ray-data-2.7.1-py39hf3d152e_1.conda
linux-64::ray-data-2.8.0-py310hff52083_0.conda
linux-64::ray-data-2.1.0-py38h578d9bd_0.conda
linux-64::ray-data-2.6.2-py310hff52083_0.conda
linux-64::ray-data-2.8.1-py310hff52083_0.conda
linux-64::ray-data-2.7.1-py311h38be061_0.conda
linux-64::ray-data-2.9.1-py39hf3d152e_0.conda
linux-64::ray-data-2.8.0-py39hf3d152e_0.conda
linux-64::ray-data-2.6.3-py310hff52083_0.conda
linux-64::ray-data-2.8.0-py39hf3d152e_2.conda
linux-64::ray-data-2.9.1-py38h578d9bd_0.conda
linux-64::ray-data-2.2.0-py39hf3d152e_2.conda
linux-64::ray-data-2.7.0-py39hf3d152e_0.conda
linux-64::ray-data-2.9.0-py310hff52083_0.conda
linux-64::ray-data-2.0.1-py310hff52083_1.tar.bz2
linux-64::ray-data-2.8.0-py310hff52083_1.conda
linux-64::ray-data-2.9.1-py311h38be061_0.conda
linux-64::ray-data-2.8.1-py311h38be061_1.conda
linux-64::ray-data-2.4.0-py39hf3d152e_0.conda
linux-64::ray-data-2.2.0-py38h578d9bd_2.conda
linux-64::ray-data-2.0.0-py37h89c1867_0.tar.bz2
linux-64::ray-data-2.5.1-py310hff52083_0.conda
linux-64::ray-data-2.8.1-py39hf3d152e_1.conda
linux-64::ray-data-2.9.0-py311h38be061_2.conda
linux-64::ray-data-2.1.0-py310hff52083_1.conda
linux-64::ray-data-2.6.3-py38h578d9bd_0.conda
linux-64::ray-data-2.2.0-py310hff52083_1.conda
linux-64::ray-data-2.3.0-py39hf3d152e_0.conda
linux-64::ray-data-2.5.1-py39hf3d152e_0.conda
linux-64::ray-data-2.5.1-py38h578d9bd_0.conda
linux-64::ray-data-2.9.0-py38h578d9bd_2.conda
linux-64::ray-data-2.2.0-py310hff52083_0.conda
linux-64::ray-data-2.2.0-py310hff52083_2.conda
linux-64::ray-data-2.8.1-py311h38be061_0.conda
linux-64::ray-data-2.9.1-py310hff52083_0.conda
linux-64::ray-data-2.7.1-py310hff52083_1.conda
linux-64::ray-data-2.6.3-py38h578d9bd_1.conda
linux-64::ray-data-2.6.1-py310hff52083_0.conda
linux-64::ray-data-2.8.1-py38h578d9bd_1.conda
linux-64::ray-data-2.6.3-py310hff52083_1.conda
linux-64::ray-data-2.7.0-py38h578d9bd_0.conda
linux-64::ray-data-2.9.0-py39hf3d152e_0.conda
linux-64::ray-data-2.9.0-py310hff52083_2.conda
linux-64::ray-data-2.6.2-py39hf3d152e_0.conda
linux-64::ray-data-2.8.0-py311h38be061_2.conda
linux-64::ray-data-2.9.0-py39hf3d152e_1.conda
linux-64::ray-data-2.5.0-py39hf3d152e_0.conda
linux-64::ray-data-2.1.0-py39hf3d152e_0.conda
linux-64::ray-data-2.0.0-py38h578d9bd_1.tar.bz2
linux-64::ray-data-2.8.1-py39hf3d152e_0.conda
linux-64::ray-data-2.0.1-py38h578d9bd_0.tar.bz2
linux-64::ray-data-2.1.0-py310hff52083_0.conda
linux-64::ray-data-2.5.0-py38h578d9bd_0.conda
linux-64::ray-data-2.8.0-py38h578d9bd_2.conda
linux-64::ray-data-2.8.1-py38h578d9bd_0.conda
linux-64::ray-data-2.8.0-py311h38be061_0.conda
linux-64::ray-data-2.7.1-py38h578d9bd_1.conda
linux-64::ray-data-2.9.0-py39hf3d152e_2.conda
linux-64::ray-data-2.7.1-py311h38be061_1.conda
linux-64::ray-data-2.2.0-py39hf3d152e_0.conda
linux-64::ray-data-2.2.0-py38h578d9bd_1.conda
-    "pandas >=1.3",
+    "pandas >=1.3,<2.2*",
```

### CODE REVIEW

File: recipe/patch_yaml/ray-packages.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  name: ray-data
+  version_le: 2.9.1
+  timestamp_lt: 1706823371000
+then:
+  - replace_depends:
+      old: pandas?( *)
+      new: pandas >=1.3,<2.2*
+---
+if:
+  name: ray
```
Comment (apmorton) [inline recipe/patch_yaml/ray-packages.yaml:13]: is this meant to be here? afaik no `ray` package exists on conda-forge.

Comment (timkpaine) [reply recipe/patch_yaml/ray-packages.yaml]: nope, older versions didn't have a pandas version specifier so I was trying to handle that but I just hardcoded in `new`

**Comment by isuruf** [comment]
Is this ready for review?

cc @conda-forge/ray-packages

**Comment by timkpaine** [comment]
I don't think `<2.2*`  works due to a longstanding bug, I think it needs to be `<2.2.0a0` or similar

File: recipe/patch_yaml/ray-packages.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+# https://github.com/conda-forge/ray-packages-feedstock/issues/147
+if:
+  name: ray-data
+  version_le: 2.9.1
+  timestamp_lt: 1706823371000
+then:
+  - replace_depends:
+      old: pandas?( *)
+      new: pandas >=1.3,<2.2*
```
Comment (isuruf) [inline recipe/patch_yaml/ray-packages.yaml:10]:
```suggestion
  - tighten_depends:
      name: pandas
      upper_bound: 2.2.0
```

**Comment by isuruf** [comment]
Can someone from @conda-forge/ray-packages approve?

**Review by apmorton — APPROVED** [review]

**Review by apmorton — CHANGES_REQUESTED** [review]
actually, this needs to be updated to include 2.9.2 since that has been released without a recipe update to incorporate this.

File: recipe/patch_yaml/ray-packages.yaml [hunk 3]
```diff
@@ -24,3 +24,13 @@ then:
   - replace_depends:
       old: grpcio !=1.56.0
       new: grpcio !=1.56.0,<1.58.0b0
+---
+# https://github.com/conda-forge/ray-packages-feedstock/issues/147
+if:
+  name: ray-data
+  version_le: 2.9.1
+  timestamp_lt: 1706823371000
```
Comment (isuruf) [inline recipe/patch_yaml/ray-packages.yaml:32]:
```suggestion
  version_le: 2.9.2
  timestamp_lt: 1707760902000
```

**Pull request merged**